app/controllers/scim_rails/application_controller.rb

```diff
@@ -38,6 +38,26 @@ def authenticate_with_oauth_bearer
     
     # Shared stuff...
 
+    def contains_square_brackets?(string)
+      (string =~ /\[.*\]/).is_a?(Numeric)
+    end
+
+    def before_square_brackets(string)
+      string.match(/([^\[]+)/).to_s
+    end
+
+    def inside_square_brackets(string)
+      string.match(/(?<=\[).+?(?=\])/).to_s
+    end
+
+    def after_square_brackets(string)
+      string.match(/(?<=\]).*/).to_s
+    end
+
+    def inside_quotations(string)
+      string.match(/(?<=")[^"]+(?=")/).to_s
+    end
+
```

Comment:
🌵 If these are just our own helper methods, these sound like they should be ending in `?` to me.
The names make me think they're used for boolean checks, like for [predicate methods](http://pragmati.st/2012/03/24/the-elements-of-ruby-style-predicate-methods/#:~:text=A%20method%20that%20returns%20true,would%20name%20the%20method%20paid%3F).
These return `nil` or a string, right?
How are they used?